config/routes.tsx

```diff
@@ -4,21 +4,41 @@ import Home from "../src/screens/Home";
 import Products from "../src/screens/Products";
 import User from "../src/screens/User";
 import { RoutesName } from "@app/config/constants";
+import Register from "@app/src/screens/Register";
+import { getStorageData, StorageKeys } from "../utils/storage";
+import { useState, useEffect } from "react";
 
+async function checkRegistration(): Promise<boolean> {
+  const isRegistered = await getStorageData(StorageKeys.isRegistered);
+  return Boolean(isRegistered);
+}
 const Stack = createNativeStackNavigator();
-const Routes: React.FC = () => {
+const Routes = () => {
+  const [initialRouteName, setInitialRouteName] = useState("");
+
+  useEffect(() => {
```

Comment:
Instead of having the useEffect calling a function that sets a function to be called, you could call it immediately. 

`useEffect(async () => {

    const isRegistered = await checkRegistration();
    setInitialRouteName(isRegistered ? RoutesName.home : RoutesName.register);
  
}, [])()`

Also what you could do is to use Try/Catch to make your `checkRegistration` call